gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java

```diff
@@ -193,12 +192,24 @@ protected void loadObject (MapLayer layer, Element element) {
 					Polyline polyline = new Polyline(vertices);
 					polyline.setPosition(x, y);
 					object = new PolylineMapObject(polyline);
-				} else if ((child = element.getChildByName("ellipse")) != null) {
-					object = new EllipseMapObject(x, y - height, width, height);
+				} else if ((child = element.getChildByName("ellipse")) != null) {                                 
+					object = new EllipseMapObject(x +width/2, y - height/2, width, height);
 				}
 			}
 			if (object == null) {
-				object = new RectangleMapObject(x, y - height, width, height);
+                            int gid = -1;
+                            if((gid = element.getIntAttribute("gid", -1)) != -1){
+                                TiledMapTile tile= map.getTileSets().getTile(gid);
+                                TextureMapObject textureMapObject = new TextureMapObject(tile.getTextureRegion());
+                                textureMapObject.setX(x);
+                                textureMapObject.setY(y - height);
+                                textureMapObject.setScaleX(scaleX);
+                                textureMapObject.setScaleY(scaleY);
+                                textureMapObject.setRotation(element.getFloatAttribute("rotation", 0));
+                                object = textureMapObject;
+                            } else {
+                               object = new RectangleMapObject(x, y - height, width, height); 
```

Comment:
RectangleObject coordinates are not centered here and we should keep it consistent with Ellipse. I suggest you revert the changes to Ellipse.